docs/architecture/psa-thread-safety.md

```diff
@@ -67,16 +67,28 @@ In the medium to long term, performing a slow or blocking operation (for example
 
 We may want to go directly to a more sophisticated approach because when a system works with a global lock, it's typically hard to get rid of it to get more fine-grained concurrency.
 
+### Key destruction short-term requirements
+
+#### Summary of guarantees when `psa_destroy_key` returns
```

Comment:
Please use unique section titles so that there are unique anchors (the anchors are automatically derived from the section titles, with some case and punctuation normalization).